usr/src/uts/common/fs/zfs/arc.c

```diff
@@ -4988,9 +4989,16 @@ arc_read(zio_t *pio, spa_t *spa, const blkptr_t *bp, arc_done_func_t *done,
 				cb->l2rcb_bp = *bp;
 				cb->l2rcb_zb = *zb;
 				cb->l2rcb_flags = zio_flags;
+				uint64_t asize = vdev_psize_to_asize(vd, size);
```

Comment:
Variable declarations in the middle of blocks are generally frowned upon. Please move this to the top of the block.